### PR TITLE
fix Service Port Name changce Signed-off-by: taoxin 464693850@qq.com

### PR DESCRIPTION
**Pull request by cobb-tx**
**Please provide a description of this PR:**

fix https://github.com/istio/istio/issues/38785 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [  x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by cobb-tx** [comment]
and can turn off https://github.com/istio/istio/pull/38776

File: pilot/pkg/xds/endpoint_builder.go [hunk 1]
```diff
@@ -282,7 +282,7 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 			if !ep.IsDiscoverableFromProxy(b.proxy) {
 				continue
 			}
-			if svcPort.Name != ep.ServicePortName {
+			if svcPort.Port != int(ep.EndpointPort) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/endpoint_builder.go:285]: This is not right, svc port can be different from endpoint port

Comment (cobb-tx) [reply pilot/pkg/xds/endpoint_builder.go:285]: yes your are right。 i test find that

Comment (cobb-tx) [reply pilot/pkg/xds/endpoint_builder.go:285]: Do you have any idea to fix the bug？

Comment (hzxuzhonghu) [reply pilot/pkg/xds/endpoint_builder.go:285]: This is a good catcha, will investigate

Comment (cobb-tx) [reply pilot/pkg/xds/endpoint_builder.go:285]: i find a way to fix the bug？ still hava 503 for a little while .send next xds is ok.

**Pull request closed without merging**